.github/workflows/test.yaml

```diff
@@ -0,0 +1,174 @@
+
+            - name: Run internal Django tests
+              run: poetry run python website/manage.py check
+
+            - name: Test the templates
+              run: poetry run python website/manage.py templatecheck --project-only
+
+            - name: Make migrations
+              run: poetry run python website/manage.py makemigrations --no-input --check --dry-run
+
+            - name: Run coverage
+              run: poetry run coverage run website/manage.py test website/
+
+            - name: Report coverage
+              run: |
+                coverage report --fail-under=100 --omit "website/registrations/urls.py" website/registrations/**.py
+                coverage report --fail-under=100 --omit "website/payments/urls.py" website/payments/**.py
+                coverage report
+            
+            - name: Create coverage report
+              run: coverage html --directory=covhtml --title="Coverage Report"
+
+            - name: Save coverage report
+              uses: actions/upload-artifact@v1
+              with:
+                  name: coverage-html
+                  path: covhtml
+
+
+    documentation-tests:
+        name: Documentation Tests
+        runs-on: ubuntu-18.04
+        container: thalia/concrexit-dependencies
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v2
+
+            - name: Install any missing dependencies
+              run: poetry install --no-interaction --extras "docs"
+
+            - name: Run doctest
+              working-directory: docs
+              run: poetry run make doctest
+
+            - name: Run Sphinx
+              working-directory: docs
+              run: poetry run sphinx-build -W . _build -j $(nproc)
+
+            - name: Run API docs script
+              working-directory: docs
+              run: |
+                  poetry run ./generate-apidocs.sh
+                  git diff --exit-code
+
+            - name: Save documentation HTML
+              uses: actions/upload-artifact@v1
+              with:
+                  name: docs-html
+                  path: docs/_build
+
+    upload-coverage:
+        name: Deploy Coverage Report
+        runs-on: ubuntu-18.04
+        needs: [tests]
+        steps:
+            - uses: actions/download-artifact@v1
+              with:
+                  name: coverage-html
+                  path: html
+
+            - uses: rlespinasse/github-slug-action@master
+            - name: Sync files to S3
+              run: aws s3 sync --only-show-errors html "s3://thalia-coverage/${GITHUB_REF_SLUG}/"
+              env:
+                  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+                  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            
+            - name: Show coverage report link
+              run: echo "You can view the coverage report on https://s3-eu-west-1.amazonaws.com/thalia-coverage/${GITHUB_REF_SLUG}/index.html"
```

Comment:
I know this is not really related to this issue, but the simpler url to reach a S3 bucket would be `https://thalia-documentation.s3.amazonaws.com/${GITHUB_REF_SLUG}/index.html`